src/main/java/week4/flood/IslandCount.java

```diff
@@ -0,0 +1,86 @@
+package week4.flood;
+
+
+import java.util.HashSet;
+import java.util.Set;
+import week4.IslandCountI;
+
+public class IslandCount implements IslandCountI {
+
+  /*
+   * We go through each cell, if it is set (it is an island),
+   * we increase the count and reset all the tiles in that island
```

Comment:
It should be fairly easy to implement the same solution without modifying the map array